File: app/src/main/java/org/jellyfin/androidtv/ui/search/SearchActivity.java
package org.jellyfin.androidtv.ui.search;

import android.os.Bundle;
import android.speech.SpeechRecognizer;

import org.jellyfin.androidtv.ui.base.BaseActivity;

import androidx.fragment.app.Fragment;

public class SearchActivity extends BaseActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        boolean isSpeechEnabled = SpeechRecognizer.isRecognitionAvailable(this);

        // Determine fragment to use
        Fragment searchFragment = isSpeechEnabled
                ? new LeanbackSearchFragment()
                : new TextSearchFragment();

        // Add fragment
        getSupportFragmentManager()
                .beginTransaction()
                .replace(android.R.id.content, searchFragment)
                .commit();
    }

    @Override
    public boolean onSearchRequested() {
        // Reset layout
        recreate();

        return true;
    }
}

File: app/src/main/java/org/jellyfin/androidtv/ui/search/SearchActivity.kt
package org.jellyfin.androidtv.ui.search

import android.Manifest
import android.content.pm.PackageManager
import android.os.Bundle
import android.speech.SpeechRecognizer
import androidx.core.content.ContextCompat
import androidx.fragment.app.FragmentActivity
import org.jellyfin.androidtv.R

class SearchActivity : FragmentActivity(R.layout.fragment_content_view) {
	private val isSpeechEnabled by lazy {
		SpeechRecognizer.isRecognitionAvailable(this)
			&& ContextCompat.checkSelfPermission(this, Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_DENIED
	}

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)

		// Determine fragment to use
		val searchFragment = when {
			isSpeechEnabled -> LeanbackSearchFragment()
			else -> TextSearchFragment()
		}

		// Add fragment
		supportFragmentManager
			.beginTransaction()
			.replace(R.id.content_view, searchFragment)
			.commit()
	}

	override fun onSearchRequested(): Boolean {
		// Reset layout
		recreate()

		return true
	}
}
